usaspending_api/awards/tests/test_awards_idv_v2.py

```diff
@@ -0,0 +1,300 @@
+    award_1_model = {
+        "pk": 1,
+        "type": "IDV_B_B",
+        "category": "idv",
+        "piid": 1234,
+        "type_description": "INDEFINITE DELIVERY / INDEFINITE QUANTITY",
+        "description": "lorem ipsum",
+        "generated_unique_award_id": "ASST_AW_3620_-NONE-_1830212.0481163",
+        "total_subaward_amount": 12345.00,
+        "subaward_count": 10,
+        "awarding_agency": Agency.objects.get(pk=1),
+        "funding_agency": Agency.objects.get(pk=1),
+        "recipient": LegalEntity.objects.get(pk=1),
+        "place_of_performance": Location.objects.get(pk=1),
+        "latest_transaction": TransactionNormalized.objects.get(pk=1),
+    }
+
+    award_2_model = {
+        "pk": 2,
+        "type": "A",
+        "type_description": "DEFINITIVE CONTRACT",
+        "category": "contract",
+        "piid": "5678",
+        "parent_award_piid": "1234",
+        "description": "lorem ipsum",
+        "awarding_agency": Agency.objects.get(pk=1),
+        "funding_agency": Agency.objects.get(pk=1),
+        "recipient": LegalEntity.objects.get(pk=1),
+        "total_obligation": 1000,
+        "base_and_all_options_value": 2000,
+        "generated_unique_award_id": "CONT_AW_9700_9700_03VD_SPM30012D3486",
+        "latest_transaction": TransactionNormalized.objects.get(pk=2),
+        "total_subaward_amount": 12345.00,
+        "subaward_count": 10,
+    }
+    mommy.make("awards.Award", **award_1_model)
+    mommy.make("awards.Award", **award_2_model)
+
+
+@pytest.mark.django_db
+def test_no_data_idv_award_endpoint(client):
+    """Test the /v2/awards endpoint."""
+
+    resp = client.get("/api/v2/awards/27254436/", content_type="application/json")
+    assert resp.status_code == status.HTTP_200_OK
+
+
+@pytest.mark.django_db
+def test_null_awards():
+    """Test the award.nonempty command."""
+    mommy.make("awards.Award", total_obligation="2000", _quantity=2)
+    mommy.make("awards.Award", type="U", total_obligation=None, date_signed=None, recipient=None)
+
+    assert Award.objects.count() == 3
+    assert Award.nonempty.count() == 2
+
+
+@pytest.mark.django_db
+def test_award_last_updated_endpoint(client):
+    """Test the awards endpoint."""
+
+    test_date = datetime.datetime.now()
+    test_date_reformatted = test_date.strftime("%m/%d/%Y")
+
+    mommy.make("awards.Award", update_date=test_date)
+    mommy.make("awards.Award", update_date="")
+
+    resp = client.get("/api/v2/awards/last_updated/")
+    assert resp.status_code == status.HTTP_200_OK
+    assert resp.data["last_updated"] == test_date_reformatted
+
+
+@pytest.mark.django_db
+def test_award_endpoint_generated_id(client, awards_and_transactions):
+    resp = client.get("/api/v2/awards/CONT_AW_9700_9700_03VD_SPM30012D3486/", content_type="application/json")
+    assert resp.status_code == status.HTTP_200_OK
+    assert json.loads(resp.content.decode("utf-8")) == expected_response_idv
+
```

Comment:
These tests are incomplete. Your expected_response_idv is not an IDV, its a contract. Hence, you are only testing the (non-new) ContractResponse from the endpoint. You need to query the endpoint using "ASST_AW_3620_-NONE-_1830212.0481163" (or a variant if you change the name in accordance with my other comment".... You also need a test that verifies that the parent_generated_unique_award_id is correctly linked.